### PR TITLE
[extractor/pornhub] Extract `uploader_id` field

### PR DESCRIPTION
**Pull request by Lesmiscore**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Extracts `uploader_id` field (part of `/model/NAME HERE`) in PornHub videos

### CODE REVIEW

File: yt_dlp/extractor/pornhub.py [hunk 1]
```diff
@@ -457,9 +472,13 @@ def add_format(format_url, height=None):
         self._sort_formats(
             formats, field_preference=('height', 'width', 'fps', 'format_id'))
 
+        model_profile = self._parse_json(self._search_regex(
+            r'var\s+MODEL_PROFILE\s*=\s*({.+?})\s*;', webpage, 'model profile',
+            default='{}'), video_id, fatal=False)
```
Comment (pukkandan) [inline yt_dlp/extractor/pornhub.py:477]: You can use `_search_json`

File: yt_dlp/extractor/pornhub.py [hunk 2]
```diff
@@ -457,9 +472,13 @@ def add_format(format_url, height=None):
         self._sort_formats(
             formats, field_preference=('height', 'width', 'fps', 'format_id'))
 
+       
+        model_profile = self._search_json(
+            r'var\s+MODEL_PROFILE\s*=\s*', webpage, 'model profile', video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/pornhub.py:477]:
```suggestion
            r'var\s+MODEL_PROFILE\s*=', webpage, 'model profile', video_id, fatal=False)
```

File: yt_dlp/extractor/pornhub.py [hunk 3]
```diff
@@ -457,9 +472,13 @@ def add_format(format_url, height=None):
...
         video_uploader = self._html_search_regex(
             r'(?s)From:&nbsp;.+?<(?:a\b[^>]+\bhref=["\']/(?:(?:user|channel)s|model|pornstar)/|span\b[^>]+\bclass=["\']username)[^>]+>(.+?)<',
-            webpage, 'uploader', default=None)
+            webpage, 'uploader', default=None) or model_profile.get('username')
+        uploader_id = remove_start(model_profile.get('modelProfileLink'), '/model/')
```
Comment (pukkandan) [inline yt_dlp/extractor/pornhub.py:481]:
```suggestion
```

File: yt_dlp/extractor/pornhub.py [hunk 4]
```diff
@@ -488,6 +507,7 @@ def extract_list(meta_key):
         return merge_dicts({
             'id': video_id,
             'uploader': video_uploader,
+            'uploader_id': uploader_id,
```
Comment (pukkandan) [inline yt_dlp/extractor/pornhub.py:510]:
```suggestion
            'uploader_id': remove_start(model_profile.get('modelProfileLink'), '/model/'),
```

File: yt_dlp/extractor/pornhub.py [hunk 5]
```diff
@@ -198,6 +199,20 @@ class PornHubIE(PornHubBaseIE):
...
+    }, {
+        'url': 'http://www.pornhub.com/view_video.php?viewkey=ph601dc30bae19a',
+        'info_dict': {
+            'id': 'ph601dc30bae19a',
+            'uploader': 'Projekt Melody',
+            'uploader_id': 'projekt-melody',
+            'upload_date': '20210205',
+            'title': '"Welcome to My Pussy Mansion" - CB Stream (02/03/21)',
+            'thumbnail': str,
+            'duration': int,
```
Comment (pukkandan) [inline yt_dlp/extractor/pornhub.py:211]: ?

File: yt_dlp/extractor/pornhub.py [hunk 6]
```diff
@@ -457,9 +472,12 @@ def add_format(format_url, height=None):
         self._sort_formats(
             formats, field_preference=('height', 'width', 'fps', 'format_id'))
 
+       
+        model_profile = self._search_json(
```
Comment (pukkandan) [inline yt_dlp/extractor/pornhub.py:476]:
```suggestion
        model_profile = self._search_json(
```

File: yt_dlp/extractor/pornhub.py (hunk 5 above)
Comment (Lesmiscore) [reply yt_dlp/extractor/pornhub.py:211]: I was too laxy :sweat_smile:
`duration` is int in PornHubIE anyways

Comment (Lesmiscore) [reply yt_dlp/extractor/pornhub.py:211]: I was too lazy :sweat_smile:
`duration` is int in PornHubIE anyways

**Pull request merged**